app/lib/features/events/pages/event_list_page.dart

```diff
@@ -0,0 +1,224 @@
+        controller: searchTextController,
+        leading: const Padding(
+          padding: EdgeInsets.all(8.0),
+          child: Icon(Atlas.magnifying_glass),
+        ),
+        hintText: L10n.of(context).search,
+        trailing: searchValue.isNotEmpty
+            ? [
+                IconButton(
+                  onPressed: () {
+                    FocusManager.instance.primaryFocus?.unfocus();
+                    ref.read(searchValueProvider.notifier).state = '';
+                    searchTextController.clear();
+                  },
+                  icon: const Icon(Icons.clear),
+                ),
+              ]
+            : null,
+        onChanged: (value) =>
+            ref.read(searchValueProvider.notifier).state = value,
+      ),
+    );
+  }
+
+  Widget filterChipsButtons() {
+    return SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      child: Container(
+        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
+        child: Row(
+          children: [
+            FilterChip(
+              selected: eventFilterValue == EventFilters.all,
+              label: Text(L10n.of(context).all),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.all,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.ongoing,
+              label: Text(L10n.of(context).ongoing),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.ongoing,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.upcoming,
+              label: Text(L10n.of(context).upcoming),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.upcoming,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.past,
+              label: Text(L10n.of(context).past),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.past,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildEventList(List<CalendarEvent> events) {
+    final size = MediaQuery.of(context).size;
+    final widthCount = (size.width ~/ 500).toInt();
+    const int minCount = 2;
+
+    if (events.isEmpty) return _buildEventsEmptyState();
+
+    return SingleChildScrollView(
+      child: StaggeredGrid.count(
+        crossAxisCount: max(1, min(widthCount, minCount)),
+        children: [
+          for (var event in events)
+            EventItem(
+              event: event,
+              isShowSpaceName: widget.spaceId == null,
+            ),
+        ],
```

Comment:
did you test this on desktop? because [the staggered grid docs]() state that "Wrap your children with a [StaggeredGridTile](https://pub.dev/documentation/flutter_staggered_grid_view/latest/flutter_staggered_grid_view/StaggeredGridTile-class.html) to specify their size if it's different from a 1x1 tile." and I am not seeing us that ... does it work if we go for an actual grid layout?

Reply:
Yes it works fine on Desktop. Actually used StaggeredGrid for desktop specially to mange and used best of wide area.

I wanted to get rid of Slivers so that's why used `StaggeredGrid` instead of `GridView`.

Reference video for desktop.

https://github.com/user-attachments/assets/5961c852-4746-48f1-9978-2708239b2b81